client2/services/backend.ts

```diff
@@ -0,0 +1,51 @@
+import axios, { AxiosStatic } from 'axios';
+import { MealSearchResult, RecipePerMealSelected } from '../types';
+
+class BackendService {
+  private httpClient: AxiosStatic;
+  private baseUrl: string;
+  private bearerToken: string | null;
+
+  constructor() {
+    this.baseUrl = 'http://localhost:8000';
```

Comment:
We should make this configurable via constructor arguments like
```suggestion
  constructor(apiBaseUrl: string) {
    this.baseUrl = apiBaseUrl ;
```



this way we can later instantiate the class as 

```ts
const service = new BackendService('http://localhost:8000');

// or even using an environment variable
const service = new BackendService(process.env.BACKEND_API_URL);
```

Reply:
updated